litmus-portal/frontend/src/pages/ChooseAndConfigureDashboards/index.tsx

```diff
@@ -166,65 +172,83 @@ const ChooseAndConfigureDashboards: React.FC<ChooseAndConfigureDashboardsProps>
       return applicationMetadataMap;
     };
 
-    const getSelectedDsURL = (selectedDsID: string) => {
-      const dsList: ListDataSourceResponse[] =
-        dataSourceList?.ListDataSource ?? [];
-      let selectedDsURL: string = '';
-      dsList.forEach((ds) => {
-        if (ds.ds_id === selectedDsID) {
-          selectedDsURL = ds.ds_url;
-        }
-      });
-      return selectedDsURL;
-    };
-
     useEffect(() => {
-      if (configure === true) {
-        dashboardList?.ListDashboard?.forEach(
-          (dashboardDetail: ListDashboardResponse) => {
-            if (
-              dashboardDetail.db_id === selectedDashboard.selectedDashboardID
-            ) {
-              setDashboardVars({
-                ...dashboardVars,
-                id: selectedDashboard.selectedDashboardID,
-                name: dashboardDetail.db_name,
-                dataSourceType: dashboardDetail.ds_type,
-                dashboardTypeID: dashboardDetail.db_type_id,
-                dashboardTypeName: dashboardDetail.db_type_name,
-                dataSourceID: dashboardDetail.ds_id,
-                dataSourceURL: getSelectedDsURL(dashboardDetail.ds_id),
-                agentID: dashboardDetail.cluster_id,
-                information: dashboardDetail.db_information,
-                panelGroupMap: getExistingPanelGroupMap(
-                  dashboardDetail.panel_groups
-                ),
-                panelGroups: getExistingPanelGroups(
-                  dashboardDetail.panel_groups
-                ),
-                chaosEventQueryTemplate:
-                  dashboardDetail.chaos_event_query_template,
-                chaosVerdictQueryTemplate:
-                  dashboardDetail.chaos_verdict_query_template,
-                applicationMetadataMap: getApplicationMetadataMap(
-                  dashboardDetail.application_metadata_map
-                ),
-              });
-            }
-          }
-        );
+      if (
+        configure === true &&
+        dashboardList &&
+        dashboardList.ListDashboard &&
+        dashboardList.ListDashboard.length > 0
+      ) {
+        const dashboardDetail = dashboardList.ListDashboard[0];
+        setDashboardVars({
```

Comment:
Do we need a separate local state for it? Can you please check if it can be  done by directly accessing `dashboardList` ?

Reply:
It can be done but it is more expensive to access via an array, the copy of it makes it directly accessible.